File: 1015-smallest-integer-divisible-by-k/1015-smallest-integer-divisible-by-k.go
func smallestRepunitDivByK(k int) int {
	if k == 1 {
		return 1
	}
	if k%2 == 0 || k%5 == 0 {
		return -1
	}

	num := 1
	curr := 1
	firstModK := make([]int, k)
	for {
		if firstModK[num] != 0 {
			break
		}
		firstModK[num] = curr
		num = ((num * 10) + 1) % k
		curr++
	}

	return firstModK[0]
}

File: 1015-smallest-integer-divisible-by-k/1015-smallest-integer-divisible-by-k.cpp
#pragma GCC optimize("O3", "unroll-loops");
static const int __ = []() {
    ios::sync_with_stdio(0);
    cin.tie(0);
    return 0;
}();
class Solution {
   public:
    int smallestRepunitDivByK(int k) {
        if (k == 1) return 1;
        if (k % 2 == 0 || k % 5 == 0) return -1;
        vector<int> firstModK(k, -1);
        int num = 1;
        int curr = 1;

        while (true) {
            if (firstModK[num] != -1) break;
            firstModK[num] = curr;
            num = ((num * 10) + 1) % k;
            curr++;
        }

        return firstModK[0];
    }
};